### PR TITLE
Add `RetryFailedTrialCallback` to `optuna.storages.*`

### PR DESCRIPTION
**Pull request by contramundum53**
## Motivation
`RetryFailedTrialCallback` is imported in `optuna/storages/__init__.py`, but was not included in `__all__`.

## Description of the changes
Add `RetryFailedTrialCallback` to `__all__` in `optuna/storages/__init__.py`.

### CODE REVIEW

File: optuna/storages/__init__.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 
 __all__ = [
+    "RetryFailedTrialCallback",
```
Comment (toshihikoyanase) [inline optuna/storages/__init__.py:13]: The items seems to be sorted by alphabetical order. Could you place `RetryFailedTrialCallback` after `RedisStorage`?

**Review by toshihikoyanase — COMMENTED** [review]
As can be seen in the definition of [MaxTrialsCallback](https://github.com/optuna/optuna/blob/3f1e77ffb2e47762f14a54b86907a355148be4ee/optuna/study/__init__.py#L1-L13), this change looks reasonable.

I have a small comment.

File: optuna/storages/__init__.py (hunk 1 above)
Comment (contramundum53) [reply optuna/storages/__init__.py]: Fixed.

Comment (toshihikoyanase) [reply optuna/storages/__init__.py]: Thanks!

**Review by toshihikoyanase — APPROVED** [review]
LGTM!

**Review by nzw0301 — APPROVED** [review]
Nice catch! Thanks.

**Pull request merged**